typeform/typeform.py

```diff
@@ -0,0 +1,183 @@
+import time
+import copy
+import json
+import panoply
+import urllib2
+import datetime
+
+
+FETCH_LIMIT = 500
+DESTINATION = 'typeform'
+DESTINATION_POSTFIX = '_{__table}'
+BASE_URL = 'https://api.typeform.com/v1'
+DATE_PARSER_FORMAT = '%Y-%m-%dT%H:%M:%S'
+
+
+class Typeform(panoply.DataSource):
+
+    # constructor
+    def __init__(self, source, options):
+        super(Typeform, self).__init__(source, options)
+
+        if 'destination' not in source:
+            source['destination'] = DESTINATION
+
+        # append the destination postfix, which represent the form name
+        source['destination'] += DESTINATION_POSTFIX
+
+        # configure incremental pulling
+        self._incval = None
+        if (source.get('inckey')):
+            incval = source.get('incval')
+            self._incval = (incval if incval else
+                source.get('lastTimeSucceed') or source.get('lastRuntime'))
+
+            # validate that's the incremental value is a valid
+            # date that could be parsed to a Unix timestamp before
+            # including it in the process
+            self._incval = getTimestamp(self._incval)
+
+
+        forms = source.get('forms')
+        forms = forms if forms else []
+
+        # add an 'offset' attribute used for pagination for each
+        # different form, used as the actual number to use as offset
+        self._forms = map(lambda form: {
```

Comment:
Why not be straight forward and simple... you're "rebuilding" the forms list to add a `offset` key..
```python
self._forms = forms
for f in self._forms:
    f['offset'] = 0
```